### ISSUE TITLE
The value has been taken Start with letters and no space allowed

### ISSUE THREAD

**Issue by zaferayan** [Kooboo/Kooboo#12]
Hi!
Im using Kooboo Windows version. When i try to create new web site, validation keeps showing error messages like this:
![chrome_2019-03-01_16-03-17](https://user-images.githubusercontent.com/1326963/53639842-75498400-3c3b-11e9-8e2c-d5382c196221.png)

How can i solve this problem? 
Regards,

**Comment by koobooteam** [comment]
Hi, 

I remember once I hit this problem as well. I am not able to reproduce it any more. 

Try below two things first,  the problem should go away. If not, come back again,  I will show you how to gather network data for me to debug. 

1. upgrade to the latest version, you can do that by below screen..

![kooboostart_upgrade](https://user-images.githubusercontent.com/4032195/53644507-b8bee500-3c71-11e9-9868-c2b9a854d35b.png)

2. Clear all your local storage. Do it from your chrome browser as below screenshot. 

![kooboo_clear_local_storage](https://user-images.githubusercontent.com/4032195/53644547-d0966900-3c71-11e9-9c67-68df1b9825d7.png)

**Comment by ozcanzaferayan** [comment]
Thanks for your fast response.
I tried these steps: 
1. Kooboo.exe app seems up to date. My version is 1.2.6994.42288
2. Cleared Local storage cache.  

But didnt work. 
I just saw `isUniqueName` network request. And it returns error response like "Api method Not Found"
![image](https://user-images.githubusercontent.com/1326963/53658408-c4a6a900-3c69-11e9-8136-bc5f40ef07dc.png)

**Comment by koobooteam** [comment]
Hi,

It seems like you are using two versions. From your two screenshot. One is using *.kooboo domain, one is using *.localkooboo.com. 

*.localkooboo.com by default is for Linux version or versions that does not have access to Hostfile.

Which version are you using?  
Can you show me the full request Url that return that error message?

**Comment by ozcanzaferayan** [comment]
I run site on localhost:81 
![image](https://user-images.githubusercontent.com/1326963/53659153-d7ba7880-3c6b-11e9-9d6f-867a872213d2.png)

**Comment by koobooteam** [comment]
I started from port 81, and try again, everything still works very ok for me.  I had this problem only once before but it never come back. 

Do you think we can do a remote desktop session? I am from China, not sure if the connection will be fast enough. Do you have QQ installed? the chat tool from tencent?  or teamviewer?

![kooboo_isuniquename](https://user-images.githubusercontent.com/4032195/53659441-8a073580-3c96-11e9-88fc-349f785f3c5c.png)

**Comment by koobooteam** [comment]
"Api Method Not Found" only return when it does not have find the API, that means the Site dll is not loaded. This should not happen, otherwise you were not be able to login. 

We can figure it out. We need to debug at your side, using tools like dnSpy. Do you use dnSpy?

**Comment by koobooteam** [comment]
I saw that we pack every dll into kooboo.exe that makes dnSpy not able to debug.

If you would like to debug and see what went wrong, you may use attach file with dnSpy to debug into the request.  replace your current Kooboo.exe with files from attached zip. 

I do not know what went wrong now and can not reproduce any error.  We might still able to help with remote desktop. 

[kooboo_debug_exe.zip](https://github.com/Kooboo/Kooboo/files/2920791/kooboo_debug_exe.zip)

**Comment by koobooteam** [comment]
Below is how you can debug. 

![kooboo_dnspy](https://user-images.githubusercontent.com/4032195/53661260-e5d3bd80-3c9a-11e9-9de3-2c72c6f97c93.png)

**Comment by ozcanzaferayan** [comment]
I just heard dnSpy by you from you right now. I replaced files. And will try dnSpy. 

Currently this problem occurs my corporate Windows Computer and my coworker's windows computer. 

I tried Kooboo new version right now on my personal laptop and it works fine. 

Maybe old kooboo project files can cause this issue. We will see after dnSpy inspection.

**Comment by ozcanzaferayan** [comment]
I traced on dnSpy. When i click "Create Blank Site", it detects Available method and breakpoint activates. Its good. 

After i clicked that button, "Create a New Site" Page shows. Then i filled Site Name and Domain Name fields and clicked submit button. But breakpoint not activated and showing same "The value has been taken" on site. 

Where i put my breakpoint to detect `http://local.kooboo/_api/Site/isUniqueName?SiteName=asdasd23` URL request? 
![image](https://user-images.githubusercontent.com/1326963/53665311-1fe19700-3c7c-11e9-8206-92e231e140f5.png)

**Comment by koobooteam** [comment]
Correct. Because it does not find the API, so it returns the Error of "Api Method Not Found".  That is what we need to figure out. 

If the problem is because of old Kooboo files, you can download a new version and copy the "AppData" folder to the new version folder. Everything will work the same. 

However, if possible, I would like to ask your help to figure out what the problem is so that I might fix it. 

You need to set your breakpoint at the begin request of API under "ApiMiddleWare" as below screen. 

![kooboo_api_breakpoint](https://user-images.githubusercontent.com/4032195/53676456-7f23c380-3cdd-11e9-9e18-aa0051a5804d.png)

**Comment by koobooteam** [comment]
Correction, set the breakpoint  a few line below so that it does not get hit by non-api request. 

![kooboo_api_breakpoint_down](https://user-images.githubusercontent.com/4032195/53676490-fb1e0b80-3cdd-11e9-87d3-f1a691d47de7.png)

**Comment by ozcanzaferayan** [comment]
> If the problem is because of old Kooboo files, you can download a new version and copy the "AppData" folder to the new version folder. Everything will work the same.

No im using fresh new dotnet Kooboo version.

> Correction, set the breakpoint a few line below so that it does not get hit by non-api request.  

It hits API request and returns 200. But message attribute of `jsonBody` contains  ` Api method Not Found`.

![image](https://user-images.githubusercontent.com/1326963/53714470-925e9c80-3e5f-11e9-85d7-49458b24f987.png)

**Comment by ozcanzaferayan** [comment]
I have inspected the code further. And found "Site.IsUniqueName" of `ApiMethodManager.List` looks null
![image](https://user-images.githubusercontent.com/1326963/53715052-a60b0280-3e61-11e9-8df3-ea0f45654a26.png)

![image](https://user-images.githubusercontent.com/1326963/53715156-0306b880-3e62-11e9-8468-61d073d8c4ff.png)

**Comment by koobooteam** [comment]
Hi, 

Thanks for the debugging.  I am still trying to figure out what happened. Right now, I can not really see what the problem is directly. 

Can you do me one more favor? 

1. zip the entire the entire Kooboo application that contains this problem.  You may remove the AppData folder, that is the websites you build. 

2. send to my email: guoqi@kooboo.com. 

After that you may download a new release and copy the AppData folder to new download.  I do not see the problem here. 

Thanks so much. Once I got the file, I will be able to tell you what the problem is and fix it.

**Comment by ozcanzaferayan** [comment]
I think if i find which method/class that fills `ApiMethodManager.List` dictionary, we will find the problem of  `Site.IsUniqueName`'s null.

**Comment by koobooteam** [comment]
got your email.  will fix

**Comment by ozcanzaferayan** [comment]
That's good news.
Have you experienced problem that causes this issue?
